cmd/sonobuoy/app/cp.go

```diff
@@ -37,15 +43,64 @@ func init() {
 		&path, "path", "./",
 		"TBD: location to output",
 	)
-	// TODO: Other Options.?.?
+	cmd.Flags().String("output", "./archive", "directory to write results")
 	RootCmd.AddCommand(cmd)
 }
 
 func copyResults(cmd *cobra.Command, args []string) {
-	code := 0
-	if err := ops.CopyResults(path); err != nil {
-		errlog.LogError(errors.Wrap(err, "error attempting to copy sonobuoy results"))
-		code = 1
+	outDir, err := cmd.Flags().GetString("output")
+	if err != nil {
+		errlog.LogError(fmt.Errorf("failed to get output flag value: %v", err))
+		os.Exit(1)
+	}
+
+	// Find and load the kubeconfig.
+	loadingRules := clientcmd.NewDefaultClientConfigLoadingRules()
+	configOverrides := &clientcmd.ConfigOverrides{}
+	kubeConfig := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(loadingRules, configOverrides)
+	config, err := kubeConfig.ClientConfig()
+	if err != nil {
+		errlog.LogError(fmt.Errorf("failed to get kubernetes client: %v", err))
+		os.Exit(1)
+	}
+
+	// Setup error channel an synchronization so that all errors get reported before exiting.
+	errc := make(chan error)
+	errcount := 0
+	var wg sync.WaitGroup
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		for err := range errc {
+			errcount++
+			errlog.LogError(err)
+		}
+	}()
+
+	// Get a reader that contains the tar'd output of the directory/file we're copying.
+	reader := ops.CopyResults(config, os.Stderr, errc)
+
+	// This means we left CopyResults early and will report an error.
+	if reader == nil {
+		close(errc)
+		wg.Wait()
+		os.Exit(1)
+	}
+
+	// Extract the tar'd output into a local directory under the prefix.
+	err = ops.UntarAll(reader, outDir, prefix)
```

Comment:
I don't think we should untar, that's up to them to determine.

Reply:
This is not untaring the sonobuoy results.

We run the command `tar cf - 'path/to/results'` on the pod which will take the contents of an entire directory (which contains our tar.gz results file) and output each file to stdout which we capture with a reader. I'm sure there are other approaches, but this one seems very clean to me.

Reply:
And if we do, this should go in operations. I think `app` should just be gathering config and reporting errors.

Reply:
(each file being the 1 file, sonobuoy results and the meta log that also happens to be in the results directory)

Reply:
I suspect we may not want `ops` to write files to disk which is why I am returning a reader. The app is where we are setting useful defaults for people using the `sonobuoy` command which I assert they very much do want to write results to disk. 

However, if another program were to consume the `cp` command, they may not want to be forced into writing to disk and just want to read into memory and inspect the results there.